pkg/commands/docker.go

```diff
@@ -241,15 +235,19 @@ func (c *DockerCommand) RefreshContainersAndServices() error {
 }
 
 func (c *DockerCommand) assignContainersToServices(containers []*Container, services []*Service) {
-L:
 	for _, service := range services {
 		for _, container := range containers {
 			if !container.OneOff && container.ServiceName == service.Name {
-				service.Container = container
-				continue L
+				service.Container = append(service.Container, container)
+				continue
+			}
+			if container.Container.Labels["com.docker.swarm.service.name"] == service.Name {
```

Comment:
looks like we'd be better off setting `container.ServiceName = container.Container.Labels["com.docker.swarm.service.name"]` (if it has that label) in `GetContainers()`

Reply:
ooo, yup, good point

Reply:
this one i've changed :)